### PR TITLE
Membership termination

### LINKED ISSUE #367: Add "End Membership" form
Having a standardized and authenticated way to inform our support that a member wants to end his membership would be very helpful. For now we can do this via email, later this could be fully automated via a pycroft API.

A first quick sketch to start the discussion:

_User interface:_
- "End membership" button in the frequently used actions sends them to a form
- Fields:
  - [required] date to schedule end of membership. Due to our current fee regulations it would make sense to allow only end-of-month dates(?). Otherwise we need to make clear that any started month will result in one last fee.
  - [required] check box confirming that remaining financial credit is at least zero after the end of the membership, and giving consent that remaining credit is donated. If unchecked, send user to usual contact form to go through the refund process manually.  
  - [optional] mail address to notify that request has been processed (whX adresses will not be accesible anymore)

_Support view via mail:_
- dedicated OTRS queue, could even be separated by backend
- mail contains:
  - cancel date in subject for quick sorting
  - unique identifier for the backend
  - confirmation regarding settled account/ consent to donation
  - if confirmation requested: mail address to send confirmation to

I've to yet worked with the mail/otrs backend for SIPA, but this looks doable to me. Feedback welcome.

### PR DESCRIPTION
**Pull request by ibot3**
This PR closes #367 and #295.
This PR depends on https://github.com/agdsn/pycroft/pull/332.

With this PR it is possible for users to terminate their membership in the usersuite.
The form to complete the terminations looks like this:
Image

It is also possible to cancel a scheduled termination.

This PR also includes some improvements regarding access restrictions to some pages if not all requirements were met.

E-Mail notifications should be done by pycroft.

I like the idea of a feedback field in the termination form, but I did not implement it yet to save time.

### CODE REVIEW

File: sipa/model/pycroft/user.py [hunk 1]
```diff
@@ -299,7 +299,7 @@ def to_kib(v: int) -> int:
     return (v // 1024) if v is not None else 0
 
 
-def evaluate_status(status: UserStatus):
+def evaluate_status(status: UserStatus, user: User):
```
Comment (Georgeto) [inline sipa/model/pycroft/user.py]: Make `evaluate_status` a member function of `User`?

File: sipa/model/exceptions.py [hunk 2]
```diff
@@ -27,9 +27,21 @@ class NetworkAccessAlreadyActive(Exception):
...
+
 class DBQueryEmpty(Exception):
     pass
 
 
 class LDAPConnectionError(Exception):
     pass
+
+
+class UnkownError(Exception):
```
Comment (Georgeto) [inline sipa/model/exceptions.py]: Unk**n**ownError

File: sipa/blueprints/usersuite.py [hunk 3]
```diff
@@ -408,3 +412,119 @@ def hosting(action=None):
...
+            form.end_date.default = end_date
+    else:
+        return redirect(url_for('.terminate_membership'))
+
+    if form.validate_on_submit():
+        try:
+            current_user.terminate_membership(form.end_date.data)
+        except TerminationNotPossible:
+            flash(gettext("Beendigung der Mitgliedschaft nicht möglich!"), "error")
+        except MacAlreadyExists:
```
Comment (Georgeto) [inline sipa/blueprints/usersuite.py]: This should be UnknownError, right?

**Comment by lukasjuhrich** [comment]
Status of my Review: I'm not quite understanding where the `membership_end_date` is presented to the user in the pycroft backend. Closing once I figured that out, I will try to add clarifying comments.

- [ ] Understand where the information comes from
- [ ] Improving communication of that in the code
- [ ] Adding that information to the sample backend (should take ≤5min)

**Pull request merged**